opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -735,98 +735,95 @@ namespace Opm {
 
-                }
-                else if(!haveCompletionData) {
-                    WellCommon::StatusEnum status = WellCommon::StatusFromString( record.getItem("STATUS").getTrimmedString(0));
-                    if (status == WellCommon::StatusEnum::OPEN && !well->canOpen(currentStep)) {
-                        std::string msg =
-                                "Well " + well->name() + " where crossflow is banned has zero total rate. " +
-                                "This well is prevented from opening at " + std::to_string ( m_timeMap.getTimePassedUntil(currentStep) / (60*60*24) ) + " days";
+        for( const auto& record : keyword ) {
+            const auto& wellname = record.getItem( "WELL" ).getTrimmedString(0);
+            const auto& status_str = record.getItem( "STATUS" ).getTrimmedString( 0 );
+
+            /* if all records are defaulted or just the status is set, only
+             * well status is updated
+             */
+            if( all_defaulted( record ) ) {
+                const auto status = WellCommon::StatusFromString( status_str );
+
+                for( auto* well : getWells( wellname ) ) {
+                    if( status == open && !well->canOpen(currentStep) ) {
+                        auto days = m_timeMap.getTimePassedUntil( currentStep ) / (60 * 60 * 24);
+                        std::string msg = "Well " + well->name()
+                            + " where crossflow is banned has zero total rate."
+                            + " This well is prevented from opening at "
+                            + std::to_string( days ) + " days";
                         m_messages.note(msg);
-                        continue;
+                    } else {
+                        this->updateWellStatus( *well, currentStep, status );
                     }
-                    updateWellStatus( *well, currentStep, status );
                 }
+
+                continue;
+            }
+
+            auto defaulted = []( int x ) { return x < 0; };
+            auto maybe = []( const DeckRecord& rec, const std::string& s ) {
+                const auto& item = rec.getItem( s );
+                return item.defaultApplied( 0 ) ? -1 : item.get< int >( 0 ) - 1;
+            };
+
+            const int I  = maybe( record, "I" );
+            const int J  = maybe( record, "J" );
+            const int K  = maybe( record, "K" );
+            const int C1 = maybe( record, "C1" );
+            const int C2 = maybe( record, "C2" );
+
+            const auto status = WellCompletion::StateEnumFromString( status_str );
+
+            /*
+             * Construct the updated completion with the possible status change
+             * applied. Status is changed when a completion matches all the IJK
+             * criteria *and* the completion number. A defaulted field always
+             * matches the property in question.
+             */
+            auto updated = [=]( const Completion& completion ) -> Completion {
```

Comment:
I did that at first, but you still need to get the value (I added the `defaulted` function for readability more than anything), and I figured it was clearer what was *actually* going on this way. A bit less clunky since looking up individual items from a `DeckRecord` is somewhat clunky. Oh, and the record is reused for many completions.